enhancements/sig-architecture/81-addon-lifecycle/README.md

```diff
@@ -0,0 +1,669 @@
+
+## Summary
+
+This proposal is to propose an enhancement of the `ClusterManagementAddon` and `ManagedClusterAddOn` API so that users can do
+canary upgrade on a specific addon.
+
+## Motivation
+
+Today, the addon agent is controlled by the addon manager, and when the addon manager upgrades all the agents will upgrade automatically.
+In some cases, we are seeing requirements that the user would like to rolling upgrade an addon to the new version. That is, 
+upgrade the addon on a group of clusters first, if everything works fine, then upgrade all the clusters to the new version.
+
+### Goals
+
+- Enhance the existing `ClusterManagementAddon` and `ManagedClusterAddOn` API, to support version-based addon installation 
+on a group of clusters. 
+- Enhance the existing `ClusterManagementAddon` and `ManagedClusterAddOn` API, to reflect the addon installation, upgrade 
+and rollback progress, including the addon version, progress, and completed time.
+
+### Non-Goals
+How to enhance the existing addon framework and addon manager to support multiple version addon installtion is not this 
+proposal's scope.
+
+## Proposal
+
+### Use cases
+
+#### Story 1
+- As a user, I can configure a desired version of my addon to install for a given managed cluster. 
+- I can know the installation status, including the addon version(desired version, current version), 
+progress (installing, failed, succeeded) and completed time..
+
+#### Story 2
+- As a user, I can configure a desired version of my addon to upgrade for a given managed cluster. 
+- I can know the upgrade status, including the addon version(desired version, current version), 
+progress (upgrading, failed, succeeded) and completed time..
+
+#### Story 3
+- As a user, I can configure a desired version of my addon to rollback for a given managed cluster. 
+- I can know the rollback status, including the addon version(desired version, current version), 
+progress (rolling back, failed, succeeded) and completed time.
+
+### Risks and Mitigation
+N/A
+
+## Design Details
+The design is based on [Addon install strategy #77](https://github.com/open-cluster-management-io/enhancements/pull/77/files). 
+
+### API changes
+
+```golang
+// ClusterManagementAddOnStatus represents the current status of cluster management add-on.
+type ClusterManagementAddOnStatus struct {
+  // SupportedVersions lists all the valid addon versions. 
+  // +optional
+  SupportedVersions []string `json:"supportedVersions,omitempty"`
+
+  // DefaultVersion lists the default addon versions to be installed. 
+  // +optional
+  DefaultVersion string `json:"defaultVersion,omitempty"`
+}
+```
+
+```golang
+// ManagedClusterAddOnSpec defines the install configuration of
+// an addon agent on managed cluster.
+type ManagedClusterAddOnSpec struct {
+  // …
+  InstallNamespace string `json:"installNamespace,omitempty"`
+	
+  // installVersion represents the desired addon version to install.
```

Comment:
I think I can tell from here, but which bit of code is responsible for determining what a version means and how to move from the current version to the desired version?

Reply:
Thanks for the comments, add more description to the desiredVersion. The upgrade/rollback should be triggerd if the current version listed in the status does not match the desired version.